main.tf

```diff
@@ -0,0 +1,79 @@
+locals {
+  hostname = format("%s-bastion", var.bastion_name)
+}
+
+// Dedicated service account for the Bastion instance.
+resource "google_service_account" "bastion" {
+  account_id   = format("%s-bastion-sa", var.bastion_name)
+  display_name = "Bastion Service Account"
+}
+
+// Allow access to the Bastion Host via SSH.
+resource "google_compute_firewall" "bastion-ssh" {
+  name          = format("%s-bastion-ssh", var.bastion_name)
+  network       = var.network_name
+  direction     = "INGRESS"
+  project       = var.project_id
+  source_ranges = ["0.0.0.0/0"] // TODO: Restrict further.
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+
+  target_tags = var.tag
+}
+
+// The user-data script on Bastion instance provisioning.
```

Comment:
what is the use case of installing tinyproxy? can we make user-script as a template